packages/edit-navigation/src/store/transform.js

```diff
@@ -1,12 +1,143 @@
 /**
  * External dependencies
  */
-import { sortBy } from 'lodash';
+import { get, omit, sortBy, zip } from 'lodash';
 
 /**
  * WordPress dependencies
  */
-import { createBlock, parse } from '@wordpress/blocks';
+import { serialize, createBlock, parse } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { NEW_TAB_TARGET_ATTRIBUTE } from '../constants';
+import { addRecordIdToBlock, getRecordIdFromBlock } from './utils';
+
+/**
+ * A WP nav_menu_item object.
+ * For more documentation on the individual fields present on a menu item please see:
+ * https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/nav-menu.php#L789
+ *
+ * Changes made here should also be mirrored in packages/edit-navigation/src/store/utils.js.
+ *
+ * @typedef WPNavMenuItem
+ *
+ * @property {Object} title       stores the raw and rendered versions of the title/label for this menu item.
+ * @property {Array}  xfn         the XFN relationships expressed in the link of this menu item.
+ * @property {Array}  classes     the HTML class attributes for this menu item.
+ * @property {string} attr_title  the HTML title attribute for this menu item.
+ * @property {string} object      The type of object originally represented, such as 'category', 'post', or 'attachment'.
+ * @property {string} object_id   The DB ID of the original object this menu item represents, e.g. ID for posts and term_id for categories.
+ * @property {string} description The description of this menu item.
+ * @property {string} url         The URL to which this menu item points.
+ * @property {string} type        The family of objects originally represented, such as 'post_type' or 'taxonomy'.
+ * @property {string} target      The target attribute of the link element for this menu item.
+ */
+
+export function blockToMenuItem(
+	block,
+	menuItem,
+	parentId,
+	blockPosition,
+	menuId
+) {
+	menuItem = omit( menuItem, 'menus', 'meta', '_links' );
+	menuItem.content = get( menuItem.content, 'raw', menuItem.content );
+
+	let attributes;
+
+	if ( block.name === 'core/navigation-link' ) {
```

Comment:
This still needs to include the submenu block. Otherwise those get saved as `type: 'block'`, but instead they should directly transformable to normal menu items.
```suggestion
	if ( [ 'core/navigation-link', 'core/navigation-submenu' ].includes( block.name ) ) {
```

This change should fix a couple of issues.

First the label for a submenu block somehow ends up being changed to 'Block' after clicking the save button:
<img width="437" alt="Screenshot 2021-09-21 at 12 04 09 pm" src="https://user-images.githubusercontent.com/677833/134109998-7e1ad88f-6059-441c-869a-121b3decffa7.png">

Second, I sometimes see this save error with submenus:
<img width="451" alt="Screenshot 2021-09-21 at 12 03 31 pm" src="https://user-images.githubusercontent.com/677833/134110028-e5acca88-4405-4410-ba25-22eacf512c93.png">

Reply:
Seems like `isSupportedBlock` could do this job.

Reply:
Nice, great spot! I updated that